images/kubetest_build/build_kubetest.sh

```diff
@@ -16,27 +24,99 @@ KUBETEST_STORAGE_CONTAINER="k8s-windows"
 KUBETEST_STORAGE_BLOB="testing/kubetest/kubetest_$(date '+%Y-%m-%d-%H-%M-%S')/kubetest"
 KUBETEST_STORAGE_BLOB_LATEST="testing/kubetest/kubetest_latest/kubetest"
 
+GINKGO_PARALLEL=${GINKGO_PARALLEL:-10}
+AGENT_NODES=${AGENT_NODES:-4}
+NETWORK_PLUGIN=${NETWORK_PLUGIN:-"azure"}
+
+AZ_RG_NAME=${JOB_NAME}-${PROW_JOB_ID}
+AZ_DEPLOYMENT_NAME=prow-${PROW_JOB_ID}
+
+ACS_API_MODEL_FILES=("kubernetes.json" "apimodel.json")
+ACS_API_MODEL_SENSITIVE_KEYS=("secret" "clientId" "keyData" "clientPrivateKey" "caCertificate" "etcdServerPrivateKey" \
+                              "apiServerCertificate" "clientCertificate" "etcdClientPrivateKey" "etcdServerCertificate" \
+                              "caPrivateKey" "etcdClientCertificate" "etcdPeerCertificates" "etcdPeerPrivateKeys" "kubeConfigPri      vateKey" \
+                              "apiServerPrivateKey" "kubeConfigCertificate")
+
+function redact_file {
+    # redact sensitive information from the logs ( i.e clientSecret / clientId etc )
+    for key in ${ACS_API_MODEL_SENSITIVE_KEYS[@]}; do
+          sed -i "/\"${key}\": \[/,+2d" $1
+          sed -i "/${key}/d" $1
+    done
+}
+
+function copy_acs_engine_logs {
+    # we use a regex here, not really pretty, but it will work since we know for a fact it's the only dir to match.
+    # kubetest generates logs in a tempdir with the form acs[0-9]+
+    # we use \+ instead of + because sed type regex
+    acs_folders=$(find $HOME -regextype sed -regex "$HOME/acs[0-9]\+")
+    for folder in $acs_folders; do
+        pushd $folder
+        for file in ${ACS_API_MODEL_FILES[@]}; do
+                # first redact, then copy. If redating fails for some reason, the logs will end up on the server in clear
+                # since pushing logs is automatic on exit
+                redact_file $file
+                cp $file $OUTPUT_DIR
+        done
+    done
+    popd
```

Comment:
nit picking: popd should be in the first for, since if by some chance there are actually two acs folders, after exiting this function we won't be in the dir we want. But this is just nit picking, there will always be, of course only one acs folder per container.